### PR TITLE
Add cookie customizer to CookieRequestCache

### PR DESCRIPTION
**Pull request by fb64**
Issue https://github.com/spring-projects/spring-security/issues/15204

### CODE REVIEW

File: web/src/main/java/org/springframework/security/web/savedrequest/CookieRequestCache.java [hunk 1]
```diff
@@ -152,4 +157,13 @@ public void setRequestMatcher(RequestMatcher requestMatcher) {
 		this.requestMatcher = requestMatcher;
 	}
 
+	/**
+	 * Sets the {@link Consumer}, allowing customization of cookie.
+	 * @param cookieCustomizer customize for cookie
```
Comment (marcusdacoregio) [inline web/src/main/java/org/springframework/security/web/savedrequest/CookieRequestCache.java:162]: Can you please add `@since 6.4`?

**Review by marcusdacoregio — CHANGES_REQUESTED** [review]
Thanks for the PR @fb64. I've left a single feedback inline.

Would you please add the same support in `CookieServerRequestCache`? I believe that the only difference is that the `cookieCustomizer` would be a `Consumer<ResponseCookie.Builder>`

**Comment by fb64** [comment]
> Thanks for the PR @fb64. I've left a single feedback inline.
> 
> Would you please add the same support in `CookieServerRequestCache`? I believe that the only difference is that the `cookieCustomizer` would be a `Consumer<ResponseCookie.Builder>`

As far as I know `ResponseCookie` is used by reactive stack and cannot directly used in a standard `HttpServletResponse` unless refactoring the class, that's why I used a consumer of the same `Cookie` object already used. 
It's very similar to https://github.com/spring-projects/spring-security/pull/15203

**Comment by marcusdacoregio** [comment]
Hi @fb64, I'm sorry if what I said was confusing. But I mean that the reactive implementation should be a `Consumer<ResponseCookie.Builder>`. Does that make sense?

**Comment by fb64** [comment]
> Hi @fb64, I'm sorry if what I said was confusing. But I mean that the reactive implementation should use a `Consumer<ResponseCookie.Builder>`. Does that make sense?

Got it, I added it to `CookieServerRequestCache`  sorry for the misunderstanding 😅

**Comment by marcusdacoregio** [comment]
Thanks @fb64, this is now merged into `main`.

**Pull request merged**